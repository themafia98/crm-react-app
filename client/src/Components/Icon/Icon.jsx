import React from 'react';
import PropTypes from 'prop-types';
import './Icon.scss';
const Icon = ({path, go, newClass, cbRedirectHome}) => {
    return (
        <div 
            onClick = {go && cbRedirectHome}  
            className = {!newClass ? 'Icon' : 'Icon ' + newClass}>
            <img       
                className = {go ? 'go' : null} 
                src = {process.env.PUBLIC_URL + path} 
                alt = 'Icon' 
            />
        </div>
    )
};
Icon.propTypes = {
    path: PropTypes.string.isRequired,
    go: PropTypes.oneOfType([
        PropTypes.string.isRequired,
        PropTypes.bool.isRequired
    ]),
    newClass: PropTypes.string,
    cbRedirectHome: PropTypes.func

}
export default Icon;