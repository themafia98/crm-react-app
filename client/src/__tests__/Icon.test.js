import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../Components/Icon/Icon';

const IconTest = renderer.create(
    <Icon path = '/img/logo.png' />
)

test('Icon test', () => {

    let treeIconSnapshot = IconTest.toJSON();
    expect(treeIconSnapshot).toMatchSnapshot();

});