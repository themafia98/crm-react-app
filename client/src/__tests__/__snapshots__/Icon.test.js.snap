// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test 1`] = `
<div
  className="Icon"
>
  <img
    alt="icon"
    className={null}
    src="/img/logo.png"
  />
</div>
`;
